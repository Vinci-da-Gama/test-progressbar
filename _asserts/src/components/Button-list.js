import React, {
	Component
} from 'react';
import { FormGroup, Label } from 'reactstrap';

import { ConstNumbers } from '../consts/Numbers';
import ButtonsContainer from './Buttons';

export class ButtonList extends Component {
	render() {
		return (
            <FormGroup>
                <Label for="progressbAs-select" className="text-brand">
                    Add or Minus
                </Label>
                <div className="text-center equalspace-flex">
                    { this.props.btns.map((elem, idx) => (
                        <ButtonsContainer elem={elem} key={elem + idx}
                            hasSelectedPb={ this.props.hasChoosen }
                            updateBarVal={ this.props.updateBarState } />
                    )) }
                </div>
            </FormGroup>
		);
	}
}

export default ButtonList;
