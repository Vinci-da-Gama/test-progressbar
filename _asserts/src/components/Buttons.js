import React, { Component } from 'react';
import { Button } from 'reactstrap';

import { ConstNumbers } from '../consts/Numbers';

class ButtonsContainer extends Component {
    clickToUpdate() {
        this.props.updateBarVal(this.props.elem);
    }

    render() {
        return (
            <Button
                className={`height-3rem rounded-0 mb-2
                    ${this.props.hasSelectedPb !== '' ? 'btn-outline-info' : 'btn-light'}`}
                disabled={ !this.props.hasSelectedPb }
                onClick={ () => {
                    this.clickToUpdate();
                } }>
                {
                    (Math.sign(this.props.elem) === ConstNumbers.ONE) ? '+' : ''
                }
                { this.props.elem }
            </Button>
        );
    }
}

export default ButtonsContainer;
