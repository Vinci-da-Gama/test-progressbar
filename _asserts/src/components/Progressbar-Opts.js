import React from 'react';

export const PbOpts = ({ pbId, val }) => (
    <option value={ pbId }>
        {pbId}
    </option>
);
