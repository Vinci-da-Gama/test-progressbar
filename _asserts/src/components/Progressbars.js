import React, { Component } from 'react';
import { Progress } from 'reactstrap';

import ConstNumbers from '../consts/Numbers';

export class Progressbars extends Component {
    renderBars() {
        return this.props.bars.map((elem, idx) => (
            <div key={elem.pbId + idx}>
                <div className="text-center middle-percent">
                    {elem.val}%
                </div>
                <Progress animated color={`${elem.val > this.props.limit ? 'danger' : 'info'}`}
                    className="height-3rem rounded-0"
                    value={ elem.val } />
            </div>
        ));
    }

    render() {
        return (
            <div>
                { this.renderBars() }
            </div>
        );
    }
};

export default Progressbars;
