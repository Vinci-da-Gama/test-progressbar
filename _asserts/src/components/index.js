import React from 'react';
import RootRoute from '../route';

const RootApp = () => (
    <RootRoute />
);

export default RootApp;
