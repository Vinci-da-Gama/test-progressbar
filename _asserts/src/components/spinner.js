import React, { Component } from 'react';

export class Spinner extends Component {
  render() {
    return (
        <div className="loader-container">
            <div className="loader">
                Loading...
            </div>
        </div>
    );
  }
}

export default Spinner;
