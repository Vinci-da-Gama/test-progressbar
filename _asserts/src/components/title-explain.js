import React from 'react';
import { CardTitle, CardText } from 'reactstrap';

export default () => (
    <div>
        <CardTitle className="text-center fz2em">
            Progress Bar
        </CardTitle>
        <ul className="list-group list-group-flush">
            <li className="list-group-item">
                <CardText>
                    Choose progressbar via dropdown-list.
                </CardText>
            </li>
            <li className="list-group-item border-bottom">
                <CardText>
                    Add or minus value via buttons.
                </CardText>
            </li>
        </ul>
    </div>
);
