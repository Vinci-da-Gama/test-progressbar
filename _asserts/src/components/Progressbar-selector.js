import React, { Component } from 'react';
import { Input } from 'reactstrap';

import { PbOpts } from './Progressbar-Opts';

class ProgressbarSelector extends Component {
    changeProgressbar(e) {
        this.props.selectPbById(e);
    }

    render() {
        return (
            <Input type="select" id="progressbAs-select0"
                name="progressbar-select" bsSize="lg"
                className="rounded-0 normal-bdcolor-boxshadow custom-select height-3rem appear-btn"
                onChange={ (event) => {
                    this.changeProgressbar(event);
                } }>
                <option value="default" disabled={ this.props.hasSelectedPb !== '' }>
                    Please Select
                </option>
                {this.props.bars.map((item, idx) => (
                    <PbOpts key={idx} {...item} />
                ))}
            </Input>
        );
    }
}

export default ProgressbarSelector;
