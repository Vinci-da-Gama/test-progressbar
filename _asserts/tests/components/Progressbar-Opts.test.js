import React from 'react';
import { shallow } from 'enzyme';

import { PbOpts } from '../../src/components/Progressbar-Opts';
import { SingleBarObj } from '../fixtures/singleBarItem-fixture';

describe('In components/Progressbar-Opts.js Test DropDown Options: ', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<PbOpts value={ SingleBarObj.pbId } />);
    });

    it('should render options correctly.', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
