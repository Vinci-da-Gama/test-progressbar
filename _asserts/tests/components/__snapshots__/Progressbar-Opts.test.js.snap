// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`In components/Progressbar-Opts.js Test DropDown Options:  should render options correctly. 1`] = `<option />`;
